In [7]:
lib <- c("XML","RCurl","rentrez")
# install.packages(lib,, repos = 'http://cran.us.r-project.org')
library(XML)
library(RCurl)
library(rentrez)

Loading required package: bitops


# CAS-ID to CID

### fuction : CAStocids(CAS-ID)

In [21]:
CAStocids=function(query) {
  xmlresponse = xmlParse( getURL(paste(
      "http://www.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=pccompound&retmax=100&term=",query,sep="") ) )
  cids = sapply(xpathSApply(xmlresponse, "//Id"), function(n){xmlValue(n)})
  return(cids)
}

# ~~~~ example ~~~~~~
CAStocids("64318-79-2")

[1] "5282237"

# Compound to CID

In [20]:
# entrez_dbs()
## 'pccompound' is PubChem Compound
entrez_db_summary("pccompound")
# entrez_db_searchable("pccompound")

DbName: pccompound
 MenuName: PubChem Compound
 Description: PubChem Compound Database
 DbBuild: Build160521-2055m.1
 Count: 89193328
 LastUpdate: 2016/05/25 22:36

### Unique identifier (UID) back to top
Each record within a PubChem database receives a unique identifier (UID), which is an integer that remains stable even if the content of the record is modified or enhanced over time. While "UID" is the generic term that applies to all PubChem (and other NCBI database) records, the UID is sometimes referred to by other names, depending upon the specific database being used. For example:
- **Substance identifier (SID)**
    - A substance identifier (SID) is the permanent identifier for a depositor-supplied molecule. These are found in the PubChem Substance database. Each SID corresponds to a unique external registry ID provided by a PubChem data source. (The data processing section of this document provides more details about SIDs.)

- **Compound identifier (CID)**
    - A compound identifier (CID) is the permanent identifer for a unique chemical structure. These are found in the PubChem Compound database. Each stereoisomer of a compound has its own CID. It is also possible for different tautomeric forms of the same compound to have different CID's. (The data processing section of this document provides more details about CIDs.)

    Note: Although identifiers are unique within a PubChem database, the same integer can be used as an identifier in two or more different databases. For example, "2244" is a valid identifier in both the PubChem Substance and PubChem Compound database, where:


    SID: 2244 is the PubChem Substance database record for cytidylate, and 
    CID: 2244 is the PubChem Compound database record for aspirin.

In [19]:
r_search <- entrez_search(db="pccompound", term="Gemeprost")
r_search$ids
# ids is CID list when compound searched.

[1] "6917857" "6434870" "5353643" "5282237"

## Are they really different compounds? 

https://www.ncbi.nlm.nih.gov/pccompound?term=Gemeprost

## What I want as dataframe

PubChem Compound Identifier (CID)
 
 *species*  | Compound | CID | CAS-ID
  -------------  | ------------- |   -------------  | -------------
  *Fungi Agaricus Bisporu* | Gemeprost | 5282237 | 64318-79-2
  
  
  
# Problem

### Problem in Raw data (mVOC_comp.txt)
- Naphthalene, 1,2,3,5,6,7,8,8a-octahydro-1,8a-dimethyl-7-(1-methylethenyl)-, [1S-(1.alpha.,7.alpha.,8a.alpha.)]-
- 1H-3a,7-methanoazulene, 2,3,4,7,8,8a-hexahydro-3,6,8,8-tetramethyl-, [3R-(3.alpha., 3a.beta.,7.beta.,8a.alpha.)]
- 1,4-Methanoazulen-3-ol,decahydro-1,5,5,8a-tetramethyl-,[1S-1.alpha.,3.alpha.,3a.alpha.,4.alpha.,8a.alpha.)]
- ect ......



https://www.ncbi.nlm.nih.gov/pccompound?term=(e)-5-acetyl-2%2C2-dimethyl-1-(3%27-methyl-1%27%2C3%27-butadien-1%27-yl)bicyclo%5B2.1.0%5Dpentane 
= No Result problem
 